### ISSUE TITLE
Get control branch for visualization from experiment metadata

### ISSUE THREAD

**Issue by scholtzan** [mozilla/experimenter#6129]
The reference_branch can be changed via the config files living in jetstream-config. This can result in the visualization breaking since Experimenter uses a different control branch (the one initially configured when the experiment was created) than jetstream did use for the analysis (the one from the custom config).

To fix this, Jetstream will add the `reference_branch` it used for the analysis to the experiment metadata. Experimenter can use this `reference_branch` as the source of truth when creating the visualization. Experimenter already uses this metadata to get the metric descriptions.

Blocked on https://github.com/mozilla/jetstream/issues/840

cc @LZoog

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-1587)

**Comment by LZoog** [comment]
Thanks for filing this + the other ticket! I was planning on changing the title of #6120 to something like this since what's described here will fix #6120, but I think I'll cancel that one as a dupe of this instead.

**Comment by scholtzan** [comment]
This should be unblocked now. Added `reference_branch` to metadata via https://github.com/mozilla/jetstream/pull/841

The new metadata looks something like: 


```json
{
            "metrics": {
                "view_about_logins": {
                    "friendly_name": "about:logins viewers",
                    "description": "Counts the number of clients that viewed about:logins.\n",
                    "bigger_is_better": true,
                    "analysis_bases": ["enrollments"]
                },
                "my_cool_metric": {
                    "friendly_name": "",
                    "description": "",
                    "bigger_is_better": true,
                    "analysis_bases": ["enrollments"]
                }
            },
            "outcomes": {},
            "reference_branch": "b",   // <--- This got added
            "schema_version": 3         // <--- Bumped the version to 3
}
```

A few things of note here:
* I bumped the `schema_version` from 2 to 3. I want to avoid having to re-upload metadata for all experiments that have already been analysed. So this version will indicate whether `reference_branch` is expected to exist or not.
* I did re-upload the metadata for `multi-stage-onboarding-ask-and-show-to-pin` and verified that the metadata contains the correct reference branch: 
![Screen Shot 2021-08-05 at 13 28 55](https://user-images.githubusercontent.com/1239705/128417441-8173adf9-d24b-4fbd-bdd2-40183a387abe.png)

**Comment by data-sync-user** [comment]
➤ Lauren Zugai commented:

Bumping up the points on this a tad. This added one more prop we needed to pass into all of the tables on the Results page and really puts us over the edge for just switching to Context for the analysis data (which includes the jetstream metadata and will eliminate a number of other props) since we’re planning on using Context for the experiment query in #6145 / https://mozilla-hub.atlassian.net/browse/EXP-1598 ( https://mozilla-hub.atlassian.net/browse/EXP-1598|smart-link ) and related tickets. I’m using this ticket as an opportunity to do that.

**Comment by data-sync-user** [comment]
➤ Lauren Zugai commented:

Made a separate ticket for the context bits: https://mozilla-hub.atlassian.net/browse/EXP-1640 ( https://mozilla-hub.atlassian.net/browse/EXP-1640|smart-link )

Bumping points down to reflect that.

**Comment by data-sync-user** [comment]
➤ Lauren Zugai commented:

Bumping this up again 😆 - we want to also display the overrides to the user in an alert.
